src/fullscreen/fullscreen-controller.js

```diff
@@ -0,0 +1,246 @@
+    this._isFullscreenManually = false;
+    this.registerFullScreenEvents();
+  }
+
+  /**
+   * if native fullscreen mode
+   * @memberof FullScreenController
+   * @returns {boolean} - the current fullscreen state of the document
+   */
+  isNativeFullscreen(): boolean {
+    const videoElement: ?HTMLVideoElement = typeof this._player.getVideoElement === 'function' ? this._player.getVideoElement() : false;
+    return (
+      (typeof document.fullscreenElement !== 'undefined' && Boolean(document.fullscreenElement)) ||
+      (typeof document.webkitFullscreenElement !== 'undefined' && Boolean(document.webkitFullscreenElement)) ||
+      (typeof document.mozFullScreenElement !== 'undefined' && Boolean(document.mozFullScreenElement)) ||
+      (typeof document.msFullscreenElement !== 'undefined' && Boolean(document.msFullscreenElement)) ||
+      (!!videoElement && typeof videoElement.webkitDisplayingFullscreen !== 'undefined' && Boolean(videoElement.webkitDisplayingFullscreen))
+    );
+  }
+
+  /**
+   * if fullscreen mode
+   * @memberof FullScreenController
+   * @returns {boolean} - the current fullscreen state of the document
+   */
+  isFullscreen(): boolean {
+    return (
+      this.isNativeFullscreen() ||
+      //indicator for manually full screen in ios - with css flag
+      this._isFullscreenManually
+    );
+  }
+
+  /**
+   * if mobile detected, get the video element and request fullscreen.
+   * otherwise, request fullscreen to the parent player view than includes the GUI as well
+   * @param {?HTMLElement} element - element to enter fullscreen
+   * @memberof FullScreenController
+   * @returns {void}
+   */
+  enterFullscreen(element: ?HTMLElement): void {
+    const fullScreenElement = element ? element : this._player.getView();
+    if (this._player.env.os.name === 'iOS') {
+      if (this._inBrowserFullscreen) {
+        this._enterInBrowserFullscreen(fullScreenElement);
+      } else {
+        const videoElement: ?HTMLVideoElement = this._player.getVideoElement();
+        if (videoElement && typeof videoElement.webkitEnterFullScreen === 'function') {
+          videoElement.webkitEnterFullScreen();
+        }
+      }
+    } else {
+      this._requestFullscreen(fullScreenElement);
+    }
+  }
+
+  /**
+   * exit fullscreen cross platform function
+   *
+   * @memberof FullScreenController
+   * @returns {void}
+   */
+  exitFullscreen(): void {
+    if (this._player.env.os.name === 'iOS') {
+      // player will be in full screen with this flag or otherwise will be natively full screen
+      if (this._inBrowserFullscreen) {
+        this._exitInBrowserFullscreen();
+      } else {
+        const videoElement: ?HTMLVideoElement = this._player.getVideoElement();
+        if (videoElement && typeof videoElement.webkitExitFullscreen === 'function') {
+          videoElement.webkitExitFullscreen();
+        }
+      }
+    } else if (typeof document.exitFullscreen === 'function') {
```

Comment:
for consistency, like with `_requestFullscreen`, move all of this `else if`'s to `_exitFullscreen`